src/test/java/com/commercetools/sync/products/utils/ProductUpdateActionUtilsTest.java

```diff
@@ -254,23 +284,106 @@ private void assertChangeMasterVariantEmptyErrorCatcher(final String productMock
-                .sku("testSKU")
-                .key("testKey")
-                .images(imageList)
-                .build();
-
-        AddVariant addVariant = buildAddVariantUpdateActionFromDraft(draft);
-        assertThat(addVariant.getAttributes()).isSameAs(attributeList);
-        assertThat(addVariant.getPrices()).isSameAs(priceList);
-        assertThat(addVariant.getSku()).isEqualTo("testSKU");
-        assertThat(addVariant.getKey()).isEqualTo("testKey");
-        assertThat(addVariant.getImages()).isSameAs(imageList);
+    public void buildAddVariantUpdateActionFromDraft_WithAttribsPricesAndImages_ShouldBuildCorrectAddVariantAction() {
+        // preparation
+        final List<AttributeDraft> attributeList = Collections.emptyList();
+        final List<PriceDraft> priceList = Collections.emptyList();
+        final List<Image> imageList = Collections.emptyList();
+        final ProductVariantDraft draft = ProductVariantDraftBuilder.of()
+                                                                    .attributes(attributeList)
+                                                                    .prices(priceList)
+                                                                    .sku("testSKU")
+                                                                    .key("testKey")
+                                                                    .images(imageList)
+                                                                    .build();
+
+        // test
+        final List<UpdateAction<Product>> result = buildAddVariantUpdateActionFromDraft(draft);
+
+        // assertion
+        assertThat(result).hasOnlyOneElementSatisfying(action -> {
+            assertThat(action).isInstanceOf(AddVariant.class);
+            final AddVariant addVariant = (AddVariant) action;
+            assertThat(addVariant.getAttributes()).isSameAs(attributeList);
+            assertThat(addVariant.getPrices()).isSameAs(priceList);
+            assertThat(addVariant.getSku()).isEqualTo("testSKU");
+            assertThat(addVariant.getKey()).isEqualTo("testKey");
+            assertThat(addVariant.getImages()).isSameAs(imageList);
+        });
+    }
+
+    @Test
+    public void buildAddVariantUpdateActionFromDraft_WithNoAssets_BuildsNoAddAssets() {
+        // preparation
+        final ProductVariantDraft productVariantDraft = ProductVariantDraftBuilder.of()
+                                                                                  .sku("foo")
+                                                                                  .build();
+
+        // test
+        final List<UpdateAction<Product>> result = buildAddVariantUpdateActionFromDraft(productVariantDraft);
+
+        // assertion
+        assertThat(result).containsExactlyInAnyOrder(
+            AddVariant.of(null, null, "foo", true)
+        );
+    }
+
+    @Test
+    public void buildAddVariantUpdateActionFromDraft_WithOneAsset_BuildsOneAddAssetsAction() {
```

Comment:
Isn't this test included in `buildAddVariantUpdateActionFromDraft_WithMultipleAsset_BuildsMultipleAddAssetsActions`?

Reply:
Well, in this case, maybe it's kind of redundant.

Reply:
343cc17